src/engine/scenes/TestScene.cpp

```diff
@@ -0,0 +1,128 @@
+#include "TestScene.hpp"
+
+#include "components/Backpack.hpp"
+#include "components/Model.hpp"
+
+#include "util/GLDebug.hpp"
+
+#include <glm/glm.hpp>
+
+namespace lei3d {
+    TestScene::TestScene() {
+
+    }
+
+    TestScene::~TestScene() {
+
+    }
+
+    void TestScene::LoadObjects() {
+        // load shaders
+        m_MainShader = std::make_unique<Shader>("./data/shaders/transformations.vert", "./data/shaders/transformations.frag");
+
+        // load textures
+        stbi_set_flip_vertically_on_load(true);
+
+        // load mesh from obj file (EVENTUALLY WILL WANT TO USE GTLF FILES INSTEAD)
+        // NOTE: We haven't implemented transform changing component stuff yet so the backpack might render weird/be really big if you uncomment.
+        //const std::string modelPath = "data/models/backpack/backpack.obj";
+        //std::unique_ptr<Entity> backpackObj = std::make_unique<Entity>();
+        //backpackObj->AddComponent<Backpack>();
+        //Model& model = backpackObj->AddComponent<Model>();
+        //model->Init(modelPath, *m_MainShader);
+        //m_Entities.push_back(backpackObj);
+
+        std::unique_ptr<Entity> skyboxObj = std::make_unique<Entity>();
+
+        //Test Multiple Components
+        SkyBox* skybox = skyboxObj->AddComponent<SkyBox>();
+        skyboxObj->AddComponent<Backpack>();
+        std::vector<std::string> faces
+        {
+            "data/skybox/anime_etheria/right.jpg",
+            "data/skybox/anime_etheria/left.jpg",
+            "data/skybox/anime_etheria/up.jpg",
+            "data/skybox/anime_etheria/down.jpg",
+            "data/skybox/anime_etheria/front.jpg",
+            "data/skybox/anime_etheria/back.jpg"
+        };
+        skybox->Init(faces);
+        m_Entities.push_back(std::move(skyboxObj));
+
+        m_PhysicsWorld.Create();    //TODO: This is temporary. Change this to use physics components.
+    }
+
+    void TestScene::OnUpdate(float deltaTime) {
+
+    }
+
+    void TestScene::OnPhysicsUpdate(float deltaTime) {
+        m_PhysicsWorld.Step(deltaTime);
+    }
+
+    void TestScene::OnRender() {
+        //LEI_TRACE("Rendering Test Scene");
```

Comment:
also is all of this supposed to be commented? I find it funky that things are still being drawn despite all of this being commented.